### PR TITLE
twister: Treat overflows as error on integration_platforms during integration mode

### PR DESCRIPTION
**Pull request by PerMac**
Skips due to overflows on integration_platforms during integretion mode should be treated as errors.

### CODE REVIEW

**Review by hakehuang — DISMISSED** [review]

**Comment by nashif** [comment]
> The PR exposes an existing issue:
> `3/550 warp7_m4 samples/sensor/fxas21002/sample.sensor.fxas21002 ERROR FLASH overflow but is one of the integration platforms (build)`
> What should we do with it?

you need to fix it before we are able to move forward with this PR.

**Comment by PerMac** [comment]
I chose another platform from platform_allow as an integration platform. @MaureenHelm should there be an issue raised for samples/sensor/fxas21002 overflowing flash on warp7_m4 which was the integration platform?

**Comment by MaureenHelm** [comment]
> I chose another platform from platform_allow as an integration platform for overflowing samples/sensor/fxas21002. @MaureenHelm should there be an issue raised for samples/sensor/fxas21002 overflowing flash on warp7_m4 which was the integration platform?

Yes

File: scripts/pylib/twister/twisterlib/testplan.py [hunk 1]
```diff
@@ -951,3 +943,15 @@ def _create_build_dir_link(self, links_dir_path, instance):
         instance.build_dir = link_path
 
         self.link_dir_counter += 1
+
+
+def skip_to_error_if_integration(options, instance):
```
Comment (nashif) [inline scripts/pylib/twister/twisterlib/testplan.py:948]: skip_to_error is a misleading name, sounds like the function skips to error, please reconsider the function name, maybe `switch_skipped_to_error...`

**Review by nashif — CHANGES_REQUESTED** [review]

File: scripts/pylib/twister/twisterlib/testplan.py (hunk 1 above)
Comment (PerMac) [reply scripts/pylib/twister/twisterlib/testplan.py]: changed

**Review by nashif — APPROVED** [review]

File: scripts/pylib/twister/twisterlib/testplan.py [hunk 2]
```diff
@@ -952,3 +943,15 @@ def _create_build_dir_link(self, links_dir_path, instance):
         instance.build_dir = link_path
 
         self.link_dir_counter += 1
+
+
+def change_skip_to_error_if_integration(options, instance):
```
Comment (golowanow) [inline scripts/pylib/twister/twisterlib/testplan.py:948]: looks like this function might be `class TestInstance` method

**Review by hakehuang — APPROVED** [review]

File: scripts/pylib/twister/twisterlib/testplan.py (hunk 2 above)
Comment (PerMac) [reply scripts/pylib/twister/twisterlib/testplan.py:948]: @golowanow I think it is a good suggestion. I even wanted to apply this, but found the PR merged:) I assume that your comment was just a suggestion and hence no "+/-" review. Therefore, I will leave it as it is without creating a new PR (unless you object).

**Pull request merged**